File: 프로그래머스/unrated/133499. 옹알이 （2）/옹알이 （2）.js
// function solution(babbling) {
//     let count = 0
//     let dup = ['ayaaya', 'yeye', 'woowoo', 'mama']
    
//     function check(word) {
//         for (let d of dup) {
//             if (word.includes(d)) { return 1}
//         }
//         return 0
//     }
    
//     while (babbling.length) {
//         let word = babbling.pop()
//         if (check(word) === 1) { count += 1 } 
//     }

//     return count;
// }

function solution(babbling) {
    let answer = 0
    
    for (let bab of babbling){
        if (/(aya|ye|woo|ma)\1+/g.test(bab)) continue
        if (/^(aya|ye|woo|ma)+$/g.test(bab)) answer++
    }
    return answer++
}

File: 프로그래머스/unrated/133499. 옹알이 （2）/옹알이 （2）.py
def solution(babbling):
    ans = 0
    
    baby = ['aya','ye','woo','ma']
    
    rlt = []
    
    def dfs(target,prev_word):
        if len(target) == 0:
            return 1
        
        count = 0
        for sound in baby:
            if target.startswith(sound) and prev_word != sound:
                count += dfs(target[len(sound):],sound)
        return count
    
    for target in babbling:
        rlt.append(dfs(target,""))
                
    return sum(rlt)

